libs/full/async_cuda/include/hpx/async_cuda/cuda_executor.hpp

```diff
@@ -60,11 +60,9 @@ namespace hpx { namespace cuda { namespace experimental {
         template <typename... Args>
         struct dispatch_helper<cudaError_t, Args...>
         {
-            inline cudaError_t operator()(
-                cudaError_t (*f)(Args...), Args... args)
+            inline void operator()(cudaError_t (*f)(Args...), Args... args)
```

Comment:
Leaving the return value doesn't do any harm. Also (unrelated) this `operator()()` can be `const`.

Reply:
I think making it void is actually very reasonable. `check_cuda_error` throws if the "error" is not success meaning it would in any case never be returned. On top of that, if I understood correctly, HIP actually complains about the return value being unused since many of the `f`'s are marked `nodiscard`. I'm undecided yet if I like HIP's behaviour there, but it certainly doesn't hurt in the cases we have.

Reply:
Ok, makes sense. I misread the code.